app/models/page_snapshot.rb

```diff
@@ -43,4 +43,9 @@ def parent
   def blank_match_text?
     self.match_text.blank?
   end
+
+  def filename
+    parent_page = Page.find(self.page_id)
```

Comment:
This is better referenced with just `self.page`.

Reply:
Good call. Pushed another commit to simplify this.